### PR TITLE
Integrate Order progress bar

### LINKED ISSUE #392: Integrate progress math function into progress bar UI
**Description**
Integrate [the progress function](https://github.com/cowprotocol/cowswap/issues/347) into the [progress bar UI](https://github.com/cowprotocol/cowswap/pull/366)

### PR DESCRIPTION
**Pull request by ramirotw**
# Summary

Close #392 and #347 

This integrates @alongoni's [progress bar UI](https://github.com/cowprotocol/cowswap/pull/366) and @anxolin's progress bar function into the Activity details modal.
Requirements doc: https://docs.google.com/document/d/1rJPCmztsbpIi-jklCFAQIrrPusIBYmrJvswB26I8AWA/edit#heading=h.gjdgxs

image

### CODE REVIEW

**Comment by elena-zh** [comment]
@ramirotw , thanks! I was going to leave this feedback to you right now. 
Could you please also hide the progress bar for an expired order? 
expired and cancelled

**Comment by ramirotw** [comment]
> Could you please also hide the progress bar for an expired order?

done!

**Comment by elena-zh** [comment]
And another case that is not included into this PR: when a user is conencted to a SC wallet (Gnosis safe or so), we should not show the progress bar until the order's status is changed to 'Open'
image
In other words, no need to show this bar when the order is in 'Signing' state

**Review by anxolin — COMMENTED** [review]
So nice to see a working version. 

I think this feature will need to be tested a bit with users, and will likely require some refinement, because still has some weird effects. However, i was expecting this. Now with the demo, we can explorer how to refine it. 

I would avoid to merge this to develop, until we are more sure we are ready to release it. 

One quick feedback. There's a weird effect when you open the modal, it shows the progress bar in 0% and it progresses super fast to the current %. Can we show the % in the current value from the begining to avoid this effect?

**Comment by anxolin** [comment]
Not sure if this was just a small weird case, but I saw this message
image
 and then, a few seconds later it just traded. I think @elena-zh detected similar cases before, and we had an issue, but AFAIT it was solved. So it was strange to see this again.

**Comment by ramirotw** [comment]
> One quick feedback. There's a weird effect when you open the modal, it shows the progress bar in 0% and it progresses super fast to the current %. Can we show the % in the current value from the begining to avoid this effect?

Fixed by initializing the percentage value with the current calculated one

> Not sure if this was just a small weird case, but I saw this message
> image
> and then, a few seconds later it just traded. I think @elena-zh detected similar cases before, and we had an issue, but AFAIT it was solved. So it was strange to see this again.

Weird, I'm actually using the `isUnfillable` value which is derived in: https://github.com/cowprotocol/cowswap/blob/c8e523db37042daf2ab4826eece429e5bfb50973/src/custom/state/orders/utils.ts#L211-L247

File: src/custom/components/AMMsLogo/index.tsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  }
+`
+
+export function AMMsLogo() {
+  return (
+    <Wrapper>
+      <img src={Sushi} alt="AMMs Sushiswap" />
+      <img src={Oneinch} alt="AMMs 1inch" />
+      <img src={Paraswap} alt="AMMs Paraswap" />
+      <img src={uniswap} alt="AMMs Uniswap" />
```
[the remark this answers is not in the document]
Comment (ramirotw) [reply src/custom/components/AMMsLogo/index.tsx]: So for the record, which AMMs should we include per network? HoneySwap is missing there, and ParaSwap is not on GChain.

**Comment by ramirotw** [comment]
> Also, each time when I open the activity modal, I see the progress bar empty, then it is filled in. Can we save it's state somehow OR fill it in in the background in order to prevent this issue?

This should be fixed @elena-zh 

> Hey @ramirotw , I noticed, that the progress bar is moving too fast, and in few seconds (5-9) it starts to show 'too long' message, and it is weird from my perspective.
> The issue is relevant for the Rinkeby and GC networks, in the Mainnet it runs slower.

This behavior is defined in the requirements doc. The average execution time for Rinkeby and GC is set to 15 seconds, we should probably keep iterating over these numbers so it's aligned with what's actually going on. Right now the `Looking for a CoW` state for Rinkeby and GC lasts 4.95 seconds (15*0.33), the remaining 10.05 seconds is for `Finding best onchain price` and then when crossing the 15 seconds it comes the `Your order is taking longer than usual` state.

**Comment by elena-zh** [comment]
>  I think @elena-zh detected similar cases before, and we had an issue, but AFAIT it was solved. So it was strange to see this again.

@anxolin , @ramirotw this might happen when the price comes back, so not too often, but I can face this. 
Anyways, I will keep my eye on it, and if it start happening too often, I will create a separate issue for this.

File: src/custom/components/AMMsLogo/index.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (alfetopito) [reply src/custom/components/AMMsLogo/index.tsx]: On gchain we have honeyswap, sushiswap, if I'm not mistaken SWPR.
Asking internally to be sure

[the remark this answers is not in the document]
Comment (alfetopito) [reply src/custom/components/AMMsLogo/index.tsx]: There are 4 for gchain: 'Honeyswap,SushiSwap,Baoswap,Swapr'

[the remark this answers is not in the document]
Comment (ramirotw) [reply src/custom/components/AMMsLogo/index.tsx]: Thanks, @alongoni will add the missing icons

**Comment by elena-zh** [comment]
Hey @ramirotw , changes look much better now!
I have run several tests again, and found new issues: 
1. Cancel order button under the status is not clickable when 'Price out of range' message is displayed
not working

2. I have placed 2 orders one by one in Safe, and when my Activity modal was opened, I saw 2 progress bars in it
2 progress bars
Then I closed the modal, opened it once again, and the 2nd bar was no longer displayed

3. When an order goes to the 'Open' status after 'Signing' phase, progress bar appears in the 'longer than usual' state: https://watch.screencastify.com/v/4GmNF6nyOpnoqorB0JfD
I expect to see it starting with the beginning

4. Again, when connected to the Safe wallet, and the order goes to the 'prico out of range' phase, the message is replaced by 'Longer than usual'. This replacement is visible when open the activity modal. see the video: https://watch.screencastify.com/v/YZxA3zuHpCSCHZ7sQN4X
Could you please take a look into these issues? 

Thanks!

**Comment by anxolin** [comment]
tiny detail. But i think i didn't see Uniswap icon there in the liquidity we aggregate. I would definitely add it, because uni is the biggest DEX so its nice people know. the liquidty from uni is used: 
image

**Pull request merged**